lib/internal/url.js

```diff
@@ -1084,26 +1084,32 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
-// Utility function that converts a URL object into an ordinary
-// options object as expected by the http.request and https.request
-// APIs.
+/**
+ * Utility function that converts a URL object into an ordinary options object
+ * as expected by the `http.request` and `https.request` APIs.
+ * @param {URL} url
+ * @returns {Record<string, unknown>}
+ */
 function urlToHttpOptions(url) {
+  const { hostname, pathname, port, username, password, search } = url;
   const options = {
+    __proto__: null,
+    ...url, // In case the url object was extended by the user.
     protocol: url.protocol,
-    hostname: url.hostname && StringPrototypeStartsWith(url.hostname, '[') ?
-      StringPrototypeSlice(url.hostname, 1, -1) :
-      url.hostname,
+    hostname: hostname && StringPrototypeStartsWith(hostname, '[') ?
```

Comment:
If we are checking for `hostname` to exist, then the function parameter is not a URL. URL always has a setter. In this scenario, `hostname` is `null`.

Reply:
I'm confused, I don't understand what you mean or how does it relate to this PR.

Reply:
This function checks that the parameter is a URL, but if it's a URL, than `hostname` can not be null. Therefore the JSDOC on line `1090` is false. Referencing `get hostname()` of `URL` class.

Reply:
> This function checks that the parameter is a URL

It does not, the function takes a `URL` instance and returns an object, it doesn't do any check.


This line (which was there before this PR, so if something's wrong with it, it would need to be taken to its own PR) doesn't check if `hostname` exists or if it's null, it's checking if it's truthy before calling `StringPrototypeStartsWith`.